tests/e2e/esm/index.test.js

```diff
@@ -431,6 +431,57 @@ describe('e2e esm tests', async () => {
         assert(io instanceof IOWriteable);
       }
     });
+
+    // TODO: Make a vault within this test environment's context to cover this
```

Comment:
this is a larger pending action item. what we can do is create a stubbed vault on devnet to validate the behavior against